pypeit/wavecalib.py

```diff
@@ -928,21 +977,118 @@ def run(self, skip_QA=False, debug=False,
         self.arccen, self.wvc_bpm = self.extract_arcs()
 
         # Fill up the calibrations and generate QA
+        #skip_QA = True  # for debugging
+        #msgs.warn("TURN QA BACK ON!!!")
+
         self.wv_calib = self.build_wv_calib(
             self.arccen, self.par['method'],
             skip_QA=skip_QA,
             prev_wvcalib=prev_wvcalib)
 
         # Fit 2D?
         if self.par['echelle']:
+            # Assess the fits
```

Comment:
Can this control flow be moved to self.echelle_2dfit? It seems out of place in the run method. Also, please think carefully about whether these operations can be broken off into core methods that are functions or self-contained simple classes. I realize that all the modular methods are interacting for echelle wavelength calibration, but please also understand that you are carrying out core functionality in a very complicated class, whereas the model is that base classes like wavecalib are just supposed to wrap core methods.

Reply:
I agree this was too much for `run()`

But it doesn't fit into `echelle_2dfit()` which does at it says.
The code here is to redo individual echelle orders.  So, I've
moved that code out of `run()` and into an apt named method
of BuildWaveCalib.

It mainly is control flow + calls to core methods, although still
not elegant..

Someday we need to do a major refactor of `wavecalib.py`
(together with `autoid.py`).   Will be a painful day, but definitely
needed..

Reply:
Okay for now maybe just add a TODO that this control flow needs to be consolidated in the future.